sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py

```diff
@@ -57,7 +57,6 @@
 from .._job.spark_helpers import (
     _validate_compute_or_resources,
     _validate_input_output_mode,
-    _validate_spark_configurations,
```

Comment:
What's the difference between the new `_validate_spark_configurations` and this one?

Reply:
checked `_validate_spark_configurations`. Is it possible to refactor the referred function by adding a parameter to it, like `raise_error=True`.

Reply:
no to much change, but we need add custom validate fuction here, not in to_rest_object.
You can refer to doc description.

one slight change is I removed this since it's strange to modify value in valiation function
if obj.executor_instances is None:
            obj.executor_instances = obj.dynamic_allocation_min_executors